source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidRealTimeMultiplayerClient.cs

```diff
@@ -34,111 +57,692 @@ public AndroidRealTimeMultiplayerClient(AndroidJavaObject account)
+                                new RoomStatusUpdateCallbackProxy(this, listener)))
+                        using (roomConfigBuilder.Call<AndroidJavaObject>("setRoomStatusUpdateCallback",
+                            roomStatusUpdateCallback))
+                        using (roomConfigBuilder.Call<AndroidJavaObject>("setInvitationIdToAccept", invitationId))
+                        {
+                            mRoomConfig = roomConfigBuilder.Call<AndroidJavaObject>("build");
+                            mListener = listener;
+
+                            using (var task = mRtmpClient.Call<AndroidJavaObject>("join", mRoomConfig))
+                            {
+                                AndroidTaskUtils.AddOnFailureListener(
+                                    task,
+                                    e =>
+                                    {
+                                        listener.OnRoomConnected(false);
+                                        CleanSession();
+                                    });
+                            }
+                        }
+                    });
+            }
         }
 
         public void SendMessageToAll(bool reliable, byte[] data)
         {
-            // Task<Void> sendUnreliableMessageToOthers(@NonNull byte[] messageData, @NonNull String roomId)
+            if (reliable)
+            {
+                List<Participant> participants = GetConnectedParticipants();
+                foreach (Participant participant in participants)
+                {
+                    SendMessage( /* reliable= */ true, participant.ParticipantId, data);
+                }
+
+                return;
+            }
+
+            var roomStatus = GetRoomStatus();
+            if (roomStatus != RoomStatus.Active && roomStatus != RoomStatus.Connecting)
+            {
+                OurUtils.Logger.d("Sending message is not allowed in this state.");
+                return;
+            }
+
+            string roomId = mRoom.Call<string>("getRoomId");
+            using (mRtmpClient.Call<AndroidJavaObject>("sendUnreliableMessageToOthers", data, roomId)) ;
         }
 
         public void SendMessageToAll(bool reliable, byte[] data, int offset, int length)
         {
-            // Task<Void> sendUnreliableMessageToOthers(@NonNull byte[] messageData, @NonNull String roomId)
+            SendMessageToAll(reliable, Misc.GetSubsetBytes(data, offset, length));
         }
 
         public void SendMessage(bool reliable, string participantId, byte[] data)
         {
-            // Task<Integer> sendReliableMessage(@NonNull byte[] messageData, @NonNull String roomId, @NonNull String recipientParticipantId, @Nullable ReliableMessageSentCallback callback)
-            // Task<Void> sendUnreliableMessage(@NonNull byte[] messageData, @NonNull String roomId, @NonNull String recipientParticipantId)
+            var roomStatus = GetRoomStatus();
```

Comment:
Isn't status is cached in old plugin?

Reply:
In the old plugin, there were states and each state had its own room status and there were many transitions between them. 

In the current one, we need to keep mRoom variable for various reasons and it's updated in 10 different places(mostly in callback methods). Instead of caching the room status and updating it every time, I have this method that simply asks an integer to mRoom variable. Its logic is much simpler and does not cost much.